tests/test_dialects.py

```diff
@@ -693,6 +815,89 @@ def test_presto(self):
             write="presto",
         )
 
+        self.validate(
+            "MIXED_TYPE_TO_DATE_STR(x)",
+            "DATE_FORMAT(DATE_PARSE(SUBSTR(REPLACE(CAST(x AS VARCHAR), '-', ''), 1, 8), '%Y%m%d'), '%Y-%m-%d')",
+            write="presto",
+            identity=False,
+        )
+
+        self.validate(
+            "MIXED_TYPE_TO_DATE(x)",
+            "DATE_PARSE(SUBSTR(REPLACE(CAST(x AS VARCHAR), '-', ''), 1, 8), '%Y%m%d')",
+            write="presto",
+            identity=False,
+        )
+
+        self.validate(
+            "MIXED_TYPE_TO_DI(x)",
+            "CAST(DATE_FORMAT(DATE_PARSE(SUBSTR(REPLACE(CAST(x AS VARCHAR), '-', ''), 1, 8), '%Y%m%d'), '%Y%m%d') AS INT)",
+            write="presto",
+            identity=False,
+        )
+
+        self.validate(
+            "DI_ADD(x, 1, 'DAY')",
+            (
+                "CAST(DATE_FORMAT(DATE_ADD('DAY', 1, DATE_PARSE(SUBSTR(REPLACE(CAST(x AS "
+                "VARCHAR), '-', ''), 1, 8), '%Y%m%d')), '%Y%m%d') AS INT)"
+            ),
+            write="presto",
+            identity=False,
+        )
+
+        self.validate(
+            "MIXED_TYPE_DATE_DIFF('MONTH', 20220101, 20220102)",
+            (
+                "DATE_DIFF('MONTH', DATE_PARSE(SUBSTR(REPLACE(CAST(20220101 AS VARCHAR), '-', ''), 1, 8), '%Y%m%d'), "
+                "DATE_PARSE(SUBSTR(REPLACE(CAST(20220102 AS VARCHAR), "
+                "'-', ''), 1, 8), '%Y%m%d'))"
+            ),
+            write="presto",
+            identity=False,
+        )
+
+        self.validate(
+            "MIXED_TYPE_DATE_DIFF(20220101, 20220102)",
+            (
+                "DATE_DIFF('DAY', DATE_PARSE(SUBSTR(REPLACE(CAST(20220101 AS VARCHAR), '-', ''), 1, 8), '%Y%m%d'), "
+                "DATE_PARSE(SUBSTR(REPLACE(CAST(20220102 AS VARCHAR), "
+                "'-', ''), 1, 8), '%Y%m%d'))"
+            ),
+            write="presto",
+            identity=False,
+        )
+
+        self.validate(
+            "MIXED_TYPE_ADD(20220101, 1, 'DAY', 'YYYYMMDD')",
+            (
+                "CAST(DATE_FORMAT(DATE_ADD('DAY', 1, DATE_PARSE(SUBSTR(REPLACE(CAST(20220101 "
+                "AS VARCHAR), '-', ''), 1, 8), '%Y%m%d')), '%Y%m%d') AS INT)"
+            ),
+            write="presto",
+            identity=False,
+        )
+
+        self.validate(
+            "MIXED_TYPE_ADD(20220101, 1, 'DAY', 'YYYY-MM-DD')",
```

Comment:
this test doesn't seem right to me, the generic format time string should be python style so 

YYYY -> %Y

so i think you assumed the hive is the generic format, but sqlglot uses python time str as format